contentpal/src/test/java/org/dmfs/android/contentpal/references/RowSnapshotReferenceTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.android.contentpal.references;
+
+import android.content.ContentProviderOperation;
+
+import org.dmfs.android.contentpal.Predicate;
+import org.dmfs.android.contentpal.RowReference;
+import org.dmfs.android.contentpal.RowSnapshot;
+import org.dmfs.android.contentpal.SoftRowReference;
+import org.dmfs.android.contentpal.TransactionContext;
+import org.dmfs.android.contentpal.testing.answers.FailAnswer;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import static org.hamcrest.Matchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.mock;
+
+
+/**
+ * @author Marten Gajda
+ */
+@RunWith(RobolectricTestRunner.class)
+@Config(manifest = Config.NONE)
+public class RowSnapshotReferenceTest
+{
+    @Test
+    public void testPutOperationBuilder() throws Exception
+    {
+        ContentProviderOperation.Builder resultBuilder = mock(ContentProviderOperation.Builder.class, new FailAnswer());
```

Comment:
`mock(ContentProviderOperation.Builder.class, new FailAnswer());` is repeated many times making the code a bit more bloated, could we re-use it?
With the terminology mentioned earlier, mocks which are not set up for further expectations could be called dummy, so we could have a static factory method as `dummy(class)` for the code above.
For the ones which are in fact mocks, there could be another method `strictMock(class)` because that what it is, I think, a mock which fails on anything other than set up as expectation. (Not sure of the order of the invocations, 'strict' in EasyMock refers to that as well.)

Reply:
Yeah, that could work. In general I'm more in favour of moving these things to the Matcher if possible. So we don't need to deal with it in the test. But that's not always possible.